### ISSUE TITLE
Fresh Nuxt 2 with @nuxtjs/composition-api and @vueuse/core not working

### ISSUE THREAD

**Issue by Jamiewarb** [vueuse/vueuse#897]
# Description
I can't tell if this is related to #718, as the errors are different, so I thought I'd open this issue.

From a fresh install of Nuxt 2, adding the `@nuxtjs/composition-api` and `@vueuse/core` and using `useLocalStorage` results in the following error on compilation:

```
 ERROR  Failed to compile with 1 errors

This dependency was not found:

* @vue/composition-api/dist/vue-composition-api.mjs in ./node_modules/vue-demi/lib/index.mjs

To install it, you can run: npm install --save @vue/composition-api/dist/vue-composition-api.mjs
```

I've applied some workarounds but am now stuck with the error shown at the end of this issue.

# Reproduction

Here's a repo for reproduction: https://github.com/hex-digital/nuxt-vueuse
Clone then `yarn && yarn dev`

In this repo, the `main` branch is the original issue, and the `after-718` branch shows the last error, which I'll describe below.

# Attempted workarounds

I've tried some workarounds from other threads, which are below.

## https://github.com/vueuse/vue-demi/issues/76

This one seemed closest. However, their error is for `esm.js` file extension, and mine is `.mjs`. 

I tried creating an alias similar to this answer: https://github.com/vueuse/vue-demi/issues/76#issuecomment-882249216 but with the file paths changed.

<details>
  <summary>Open this to see the alias I used</summary>

  ``` 
import { resolve } from 'path'
const vueCompositionAPIFullpath = resolve("./node_modules/@vue/composition-api/dist/vue-composition-api.mjs");

config = {
  alias: {
    // workaround
    '@vue/composition-api/dist/vue-composition-api.mjs': vueCompositionAPIFullpath,
  },
}
```
</details>

Unfortunately this gives different errors:

```
 ERROR  in ./node_modules/@vueuse/core/index.mjs      friendly-errors 15:03:21

Can't import the named export 'bypassFilter' from non EcmaScript module (only default export is available)
```

These errors are similar to #718, so I continued debugging with this:

## https://github.com/vueuse/vueuse/issues/718

After applying the first fix above, to try and sort the new error, I followed the instructions in #718 and added the `extend` to build:

```
build: {
  extend(config) {
    config.module.rules.push({
      test: /\.mjs$/,
      include: /node_modules/,
      type: "javascript/auto"
    });
  }
}
```

This allowed compilation to run successfully, however it resulted in this new error in the browser when opening the page:

```
client.js?06a0:102 Error: [vue-composition-api] must call Vue.use(VueCompositionAPI) before using any function.
    at assert (webpack-internal:///./node_modules/@vue/composition-api/dist/vue-composition-api.mjs:92)
    at getVueConstructor (webpack-internal:///./node_modules/@vue/composition-api/dist/vue-composition-api.mjs:373)
    at getWatcherVM (webpack-internal:///./node_modules/@vue/composition-api/dist/vue-composition-api.mjs:1200)
    at Module.watch (webpack-internal:///./node_modules/@vue/composition-api/dist/vue-composition-api.mjs:1467)
    at watchWithFilter (webpack-internal:///./node_modules/@vueuse/shared/index.cjs:372)
    at useStorage (webpack-internal:///./node_modules/@vueuse/core/index.mjs:740)
    at useLocalStorage (webpack-internal:///./node_modules/@vueuse/core/index.mjs:1985)
    at useMyMoment (webpack-internal:///./composables/myMoment.js:16)
    at __sfc_main.setup (webpack-internal:///./node_modules/babel-loader/lib/index.js?!./node_modules/vue-loader/lib/index.js?!./node_modules/unplugin/dist/webpack/loaders/transform.js?!./pages/index.vue?vue&type=script&lang=js&:11)
    at mergedSetupFn (webpack-internal:///./node_modules/@vue/composition-api/dist/vue-composition-api.esm.js:2162)
```

This is where I'm now stuck.

In the reproduction repo, the `main` branch is the original issue, and the `after-718` branch shows this last error.

**Comment by Jamiewarb** [comment]
Ah I've just found this: https://github.com/vuejs/composition-api/pull/837

It lists a third candidate to try, which I've now done under the `fixed` branch. With all 3 workarounds in place, Nuxt is able to compile and render.

So in order to run on a fresh Nuxt 2 with composition API, all the below is necessary:

```// nuxt.config.js

import { resolve } from 'path'
const vueCompositionAPIFullpath = resolve("./node_modules/@vue/composition-api/dist/vue-composition-api.mjs");

export default {
  // alias: {},
  alias: {
    '@vue/composition-api$': '@vue/composition-api/dist/vue-composition-api.mjs',
    '@vue/composition-api/dist/vue-composition-api.mjs': vueCompositionAPIFullpath,
  },
  build: {
    extend(config) {
      config.module.rules.push({
        test: /\.mjs$/,
        include: /node_modules/,
        type: "javascript/auto"
      });
    }
  }
}
``

Took me quite a while to find all of that information 😅

**Comment by Jamiewarb** [comment]
@antfu @danielroe You bloody legends. I came back to post a client side bug I was still getting, and boom you've solved it with 0.30.0 of the @nuxtjs composition-api module.

Huge thank you for everything ❤️

**Comment by kabaluyot** [comment]
I can still replicate the error even if using the latest nuxt composition-api version.

Solved it still by this workaround: https://github.com/vueuse/vueuse/issues/897#issuecomment-961181196
